src/groupByKey.ts

```diff
@@ -2,6 +2,24 @@ type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+type InputMap<T> = {
+  [key: string]: T;
+};
+
+export function groupByKey(
+  items: InputMap<string>[], key: string,
+): GroupsMap<object> {
+  const result: GroupsMap<object> = {};
```

Comment:
Создаем пустой объект типа GroupsMap\<T>\.
```suggestion
  const result: GroupsMap<T> = {};
```
И весь твой дальнейший код будет работать исправно.